app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -572,12 +572,10 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 Callout callout = shortSelect.getCallout();
                 for (String key : callout.getResponses()) {
                     result = intent.getExtras().getString(key);
-                    if (result != null) {
-                        session.setDatum(key,result);
-                        if(!resultSet) {
-                            resultSet = true;
-                            this.searchbox.setText(result);
-                        }
+                    if (result != null && !resultSet) {
+                        resultSet = true;

```

Comment:
Will:

Please don't do this if at all possible, or if you do make sure it is noted in the PR description. We use the format to do things like collect notes for bugfixes, so including code for other areas without noting it makes pushes easy to miss.